### PR TITLE
[NETBEANS-3393] - cleanup ArrayList uncheck conversion

### PR DESCRIPTION
**Pull request by BradWalker**
There are many places in the code where an unchecked conversion warning is printed by the compiler..

   [repeat] /home/bwalker/src/netbeans/profiler/profiler.attach/src/org/netbeans/modules/profiler/attach/dialog/AttachDialog.java:146: warning: [unchecked] unchecked conversion
   [repeat]             List<RunningVM> targets = new ArrayList();
   [repeat]                                       ^
   [repeat]   required: List<RunningVM>
   [repeat]   found:    ArrayList

I've tried to remove these but there are several places where it still exists. This is because
it would require work other than being simple to fix. I'll try to fix those later.

### CODE REVIEW

File: java/websvc.saas.codegen.java/src/org/netbeans/modules/websvc/saas/codegen/java/support/SourceGroupSupport.java [hunk 1]
```diff
@@ -184,7 +184,7 @@ private static Map createFoldersToSourceGroupsMap(final SourceGroup[] sourceGrou
     private static List<SourceGroup> getTestTargets(SourceGroup sourceGroup, Map foldersToSourceGroupsMap) {
         final URL[] rootURLs = UnitTestForSourceQuery.findUnitTests(sourceGroup.getRootFolder());
         if (rootURLs.length == 0) {
-            return new ArrayList();
+            return new ArrayList<SourceGroup>();
```
Comment (lkishalmi) [inline java/websvc.saas.codegen.java/src/org/netbeans/modules/websvc/saas/codegen/java/support/SourceGroupSupport.java]: Does diamond not work here?

File: java/websvc.saas.codegen.java/src/org/netbeans/modules/websvc/saas/codegen/java/support/SourceGroupSupport.java [hunk 2]
```diff
@@ -184,7 +184,7 @@ private static Map createFoldersToSourceGroupsMap(final SourceGroup[] sourceGrou
...
         }
         List<SourceGroup> result = new ArrayList<SourceGroup>();
```
Comment (lkishalmi) [inline java/websvc.saas.codegen.java/src/org/netbeans/modules/websvc/saas/codegen/java/support/SourceGroupSupport.java]: Also I know that probably it was not a target, but diamond here?

File: ide/versioning.util/src/org/netbeans/modules/turbo/Turbo.java [hunk 3]
```diff
@@ -366,7 +366,7 @@ public Object getMonitoredKey(Object key) {
 
     public void addTurboListener(TurboListener l) {
         synchronized(listeners) {
-            List<TurboListener> copy = new ArrayList<>(listeners);
+            List<TurboListener> copy = new ArrayList<TurboListener>(listeners);
```
[the remark this answers is not in the document]
Comment (BradWalker) [reply ide/versioning.util/src/org/netbeans/modules/turbo/Turbo.java]: Have no idea.. Good catch and it's been changed..

File: java/websvc.saas.codegen.java/src/org/netbeans/modules/websvc/saas/codegen/java/support/SourceGroupSupport.java (hunk 1 above)
Comment (BradWalker) [reply java/websvc.saas.codegen.java/src/org/netbeans/modules/websvc/saas/codegen/java/support/SourceGroupSupport.java]: Yep.. It's been changed..

File: java/websvc.saas.codegen.java/src/org/netbeans/modules/websvc/saas/codegen/java/support/SourceGroupSupport.java (hunk 2 above)
Comment (BradWalker) [reply java/websvc.saas.codegen.java/src/org/netbeans/modules/websvc/saas/codegen/java/support/SourceGroupSupport.java]: No problem.. Done..

**Comment by BradWalker** [comment]
@lkishalmi  Thanks for looking at this.. I've made the changes/comments that you made.. Sometimes, when doing this work I can overlook stuff just because there are SO MANY simple changes to make..

**Comment by lkishalmi** [comment]
Well Travis is fine with these, let's merge them in!

**Pull request merged**